src/Raven.Server/Rachis/Leader.cs

```diff
@@ -783,12 +790,14 @@ private void EmptyQueue()
                 {
                     if (_running.IsRaised() == false)
                     {
-                        e = new NotLeadingException("We are no longer the leader, this leader is disposed", e);
+                        e = new NotLeadingException(leaderDisposedMessage, e);
                     }
                     foreach (var tcs in list)
                     {
                         tcs.TrySetException(e);
                     }
+
+                    _errorOccurred.TrySetResult(e);
```

Comment:
Just to be sure, this can only be set once, we are OK with this?

Reply:
Yes, that is per leader, so this is per term. If this is set, we lose the leadership anyway